samples/csharp/MicrosoftEduGraphSamples/Workflows/ModuleWorkflow.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Graph.Beta;
+using Microsoft.Graph.Beta.Models;
+using MicrosoftGraphSDK;
+using System.Resources;
+
+namespace MicrosoftEduGraphSamples.Workflows
+{
+    /// <summary>
+    /// Contains all the workflows related to Modules
+    /// </summary>
+    internal class ModuleWorkflow
+    {
+        private readonly IConfiguration _config;
+
+        public ModuleWorkflow(IConfiguration configuration)
+        {
+            this._config = configuration;
+
+            // Verify and throw exception for input values if null or empty
+            try
+            {
+                if (string.IsNullOrEmpty(_config["classId"]))
+                {
+                    throw new Exception("Missing classId please check appconfig.json file.");
+                }
+                else if (string.IsNullOrEmpty(_config["tenantId"]))
+                {
+                    throw new Exception("Missing tenantId please check appconfig.json file.");
+                }
+                else if (string.IsNullOrEmpty(_config["secret"]))
+                {
+                    throw new Exception("Missing secret please check appconfig.json file.");
+                }
+                else if (string.IsNullOrEmpty(_config["appId"]))
+                {
+                    throw new Exception("Missing appId please check appconfig.json file.");
+                }
+                else if (string.IsNullOrEmpty(_config["teacherAccount"]))
+                {
+                    throw new Exception("Missing teacherAccount please check appconfig.json file.");
+                }
+                else if (string.IsNullOrEmpty(_config["studentAccount"]))
+                {
+                    throw new Exception("Missing studentAccount please check appconfig.json file.");
+                }
+                else if (string.IsNullOrEmpty(_config["password"]))
+                {
+                    throw new Exception("Missing password please check appconfig.json file.");
+                }
+            }
+            catch (Exception ex)
+            {
+                throw;
+            }
+        }
+        /// <summary>
+        /// Workflow to create and publish the module
+        /// </summary>
+        public async Task ClassworkAsync()
+        {
+            try
+            {
+                // Get a Graph client using delegated permissions
+                var graphClient = GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
+
+                // Create a draft module.
+                var module = await MicrosoftGraphSDK.Module.CreateAsync(graphClient, _config["classId"], "Sample Module "+ DateTime.Now.ToString("dd/MM/yyyy HHmm"), "This is the first module created using Graph SDK.");
```

Comment:
change the module description to: "This Classwork module was created with Microsoft Graph SDK"